src/testing/pod_adaptive_sampling.cpp

```diff
@@ -0,0 +1,454 @@
+        else{
+            this->pcout << "ROM already computed." << std::endl;
+        }
+    }
+    return error_greater_than_tolerance;
+}
+
+template <int dim, int nstate>
+void AdaptiveSampling<dim, nstate>::updateNearestExistingROMs(const RowVectorXd& parameter) const{
+    //Assemble ROM points in a matrix
+    MatrixXd rom_points(0,0);
+    for(auto it = rom_locations.begin(); it != rom_locations.end(); ++it){
+        rom_points.conservativeResize(rom_points.rows()+1, 2);
+        rom_points.row(rom_points.rows()-1) = it->first;
+    }
+
+    //Get distances between newly added snapshots and ROM points
+    ProperOrthogonalDecomposition::MinMaxScaler scaler;
+    MatrixXd scaled_rom_params = scaler.fit_transform(rom_points);
+    RowVectorXd scaled_point = scaler.transform(parameter);
+
+    VectorXd distances = (scaled_rom_params.rowwise() - scaled_point).rowwise().squaredNorm();
+
+    std::vector<int> index(distances.size());
+    std::iota(index.begin(), index.end(), 0);
+
+    std::sort(index.begin(), index.end(),
+              [&](const int& a, const int& b) {
+                  return distances[a] < distances[b];
+              });
+
+    //For 2(n+1) nearest points, if error>tolerance, recompute adjoint 1
+    pcout << "Searching ROM points near: " << parameter << std::endl;
+    for(int i = 0 ; i < 6 ; i++){
+        pcout << "ROM point: " << rom_locations[index[i]].first << " Error: " << rom_locations[index[i]].second->total_error << std::endl;
+        if(std::abs(rom_locations[index[i]].second->total_error) > tolerance){
+            pcout << "Total error greater than tolerance. Recomputing ROM solution" << std::endl;
+            std::shared_ptr<ProperOrthogonalDecomposition::ROMSolution<dim, nstate>> rom_solution = solveSnapshotROM(rom_locations[index[i]].first);
+            std::shared_ptr<ProperOrthogonalDecomposition::ROMTestLocation < dim,nstate >> rom_location = std::make_shared<ProperOrthogonalDecomposition::ROMTestLocation < dim, nstate>>(rom_locations[index[i]].first, rom_solution);
+            rom_locations[index[i]] = std::make_pair(rom_locations[index[i]].first, rom_location);
+        }
+    }
+}
+
+template <int dim, int nstate>
+dealii::LinearAlgebra::distributed::Vector<double> AdaptiveSampling<dim, nstate>::solveSnapshotFOM(const RowVectorXd& parameter) const{
+    this->pcout << "Solving FOM at " << parameter << std::endl;
+    Parameters::AllParameters params = reinitParams(parameter);
+
+    std::unique_ptr<FlowSolver::FlowSolver<dim,nstate>> flow_solver = FlowSolver::FlowSolverFactory<dim,nstate>::select_flow_case(&params, parameter_handler);
+
+    // Solve implicit solution
+    auto ode_solver_type = Parameters::ODESolverParam::ODESolverEnum::implicit_solver;
+    flow_solver->ode_solver =  PHiLiP::ODE::ODESolverFactory<dim, double>::create_ODESolver_manual(ode_solver_type, flow_solver->dg);
+    flow_solver->ode_solver->allocate_ode_system();
+    flow_solver->ode_solver->steady_state();
```

Comment:
same comment as before about just calling `flow_solver-run()`

Reply:
Fixed in most recent commit.